api/simqueue/db.py

```diff
@@ -173,31 +298,271 @@ async def query_jobs(
             filters.append(jobs.c.timestamp_submission >= date_range_start)
     elif date_range_end:
         filters.append(jobs.c.timestamp_submission <= date_range_end)
-
+    compiled_query= ""
     if fields is None:
         select = jobs.select()
     else:
-        select = jobs.select(*[literal_column(field) for field in fields])
+        #list_fields =["jobs.c."+field for field in fields]
+        list_fields =["jobs.c."+field for field in fields]
+        ####
+        #list_fields =[text(field) for field in fields]
+        
+        #select = jobs.select().with_only_columns(*[Column(field) for field in fields])
+        select = jobs.select().with_only_columns(*[literal_column(field) for field in fields])
+        ###
+        
+        #select = jobs.select().with_only_columns([jobs.c.timestamp_completion])
+        #select = jobs.select(*[literal_column(field) for field in fields])
+        compiled_query=str(select.compile(compile_kwargs={"literal_binds": True}))
     if filters:
+        compiled_query=str(select.compile(compile_kwargs={"literal_binds": True}))
         query = select.where(*filters).offset(from_index).limit(size)
+        #compiled_query=str(query.compile(compile_kwargs={"literal_binds": True}))
+        #print(compiled_query)
     else:
         query = select.offset(from_index).limit(size)
-
+    compiled_query=str(query.compile(compile_kwargs={"literal_binds": True}))
     results = await database.fetch_all(query)
+    #print(results)
+    #compiled_query=str(results.compile(compile_kwargs={"literal_binds": True}))
+    #return(results)
+    #return [dict(result) for result in results]
+    if fields:
+        return [dict(result) for result in results]
     return [transform_fields(await follow_relationships(dict(result))) for result in results]
 
 
 async def get_job(job_id: int):
     query = jobs.select().where(jobs.c.id == job_id)
     result = await database.fetch_one(query)
+    if result is not None:
+        intermediate_result =dict(result)
+        return transform_fields(await follow_relationships(intermediate_result))
+    else:
+        raise HTTPException(status_code=status_codes.HTTP_404_NOT_FOUND,
```

Comment:
I think we should avoid raising `HTTPException` inside `db.py`, as this mixes user interaction with database access. Instead `get_job` should return None if the job doesn't exist, and the exception should be raised in `queue.py`